multimap/src/main/java/org/infinispan/multimap/impl/EmbeddedMultimapCache.java

```diff
@@ -91,7 +93,13 @@ public EmbeddedMultimapCache(Cache<K, Collection<V>> cache) {
    @Override
    public CompletableFuture<Optional<CacheEntry<K, Collection<V>>>> getEntry(K key) {
       requireNonNull(key, "key can't be null");
-      return CompletableFuture.supplyAsync(() -> Optional.ofNullable(cache.getAdvancedCache().getCacheEntry(key)));
+      return CompletableFuture.supplyAsync(() -> {
```

Comment:
I know you didn't write this, but this is wrong. This should be using the explicit executor method. It looks like the checkstyle wasn't properly looking for `supplyAsync` :( https://github.com/infinispan/infinispan/blob/master/checkstyle/src/main/resources/checkstyle.xml#L125. I created https://issues.jboss.org/browse/ISPN-10800 to handle that.

Reply:
Thanks